### PR TITLE
Add M1 workflow

### PR DESCRIPTION
**Pull request by aquacash5**
- Add workflows for apple's M1 platform
- Fix issue with core builds

### CODE REVIEW

File: .github/workflows/aws-upload-windows.yml [hunk 1]
```diff
@@ -1,101 +1,76 @@
...
-        
-
-    - name: Install Qt
-      if: steps.cache-qt.outputs.cache-hit != 'true'
-      run: |
-        pip install aqtinstall
-        python -m aqt install --outputdir $env:QT_BASE_DIR --base $env:QT_BASE_URL $env:QT_VERSION windows desktop ${{ steps.qt-setup.outputs.QT_ARCH }}
+      - uses: actions/checkout@v2
 
+      - name: Set up QT variables
```
Comment (SerhiiGadzhilov) [inline .github/workflows/aws-upload-windows.yml:25]: As I can this task doesn't setup QT variables anymore.
Maybe it's better to move this code into `"Install OpenSSL"`
Also as I can see we have only "x64" build hence we can remove the following condition: `if ('${{ matrix.arch }}' -eq 'x64')`

File: .github/workflows/aws-upload-mac.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      matrix:
+        include:
+          - os: "macos-10.15"
+            target: "10.14"
+            shell: "bash"
+            destination: macOS_x64
+          - os: [self-hosted, macOS, arm64]
+            target: ""
+            shell: "/usr/bin/arch -arch arm64e /bin/bash --noprofile --norc -eo pipefail {0}"
+            destination: macOS_arm64
```
Comment (SerhiiGadzhilov) [inline .github/workflows/aws-upload-mac.yml:25]: As I can see .github/workflows/aws-upload-mac-10.13.yml was removed and .github/workflows/aws-upload-mac.yml doesn't have macOS10.13.
We don't need it anymore?

File: src/cmd/CMakeLists.txt [hunk 3]
```diff
@@ -14,11 +14,7 @@
...
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+add_subdirectory(synergyd)
 add_subdirectory(synergyc)
 add_subdirectory(synergys)
 add_subdirectory(syntool)
-
-if (SYNERGY_BUILD_LEGACY_SERVICE)
-    add_subdirectory(synergyd)
-endif (SYNERGY_BUILD_LEGACY_SERVICE)
```
Comment (SerhiiGadzhilov) [inline src/cmd/CMakeLists.txt:23]: As I can see we still have the option `SYNERGY_BUILD_LEGACY_SERVICE` inside `CMakeLists.txt` but after this change it does nothing. Looks like we need to remove the option or revert changes in `src/cmd/CMakeLists.txt`.

**Review by SerhiiGadzhilov — CHANGES_REQUESTED** [review]

File: .github/workflows/aws-upload-mac.yml (hunk 2 above)
Comment (aquacash5) [reply .github/workflows/aws-upload-mac.yml:25]: For the core builds, I don't think we need macOS 10.13 builds.

File: src/cmd/CMakeLists.txt (hunk 3 above)
Comment (aquacash5) [reply src/cmd/CMakeLists.txt:23]: You are correct. I will make that change

File: .github/workflows/aws-upload-windows.yml (hunk 1 above)
Comment (aquacash5) [reply .github/workflows/aws-upload-windows.yml]: Until we need a 32-bit version, I will set the action to only make a 64-bit version.

**Review by SerhiiGadzhilov — APPROVED** [review]

**Comment by SerhiiGadzhilov** [comment]
@aquacash5 Could you please, check what is wrong with actions after merging this PR
5A9AD548-CADD-4D8A-9642-1A4CA4A8411E
:

**Pull request merged**